core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/GoogleProjectRegistrationComponent.scala

```diff
@@ -73,6 +72,22 @@ trait GoogleProjectRegistrationComponent {
           DBIO.successful(Option(GoogleProjectRegistrationRecord.toGoogleProjectRegistration(googleProjectRegRec)))
       }
 
+    def findByIdsAndBillingProject(
+      googleProjectIds: Set[GoogleProjectId],
+      billingProjectName: Option[RawlsBillingProjectName],
+      pageSize: Int,
+      offset: Int
+    ): ReadAction[Seq[GoogleProjectRegistration]] = {
+      val query = googleProjectRegistrationQuery.filter(_.googleProjectId inSet googleProjectIds.map(_.value))
+      val filteredQuery = billingProjectName match {
+        case Some(name) => query.filter(_.billingProject === name.value)
```

Comment:
maybe a little nicer, up to you
```suggestion
        case Some(RawlsBillingProjectName(name)) => query.filter(_.billingProject === name)
```